source/extensions/filters/network/thrift_proxy/protocol.h

```diff
@@ -73,6 +73,15 @@ class Protocol {
    */
   virtual bool readMessageEnd(Buffer::Instance& buffer) PURE;
 
+  /**
+   * Peeks the start of a Thrift protocol reply payload in the buffer and updates the reply
+   * type parameter with the reply type of the payload.
+   * @param buffer the buffer to peek from
+   * @return true if reply type was successfully read, false if more data is required
+   * @throw EnvoyException if the data is not a valid payload
+   */
+  virtual bool peekReplyPayload(Buffer::Instance& buffer, ReplyType& reply_type) PURE;
```

Comment:
could this be body const?

Reply:
it could be but that opens a can of worms on the peek buffer helpers we have. Ill look into fixing that in a later PR i think.